Support TIN in dist2d calculations

Komzpa opened the pull request.

pramsey commented:
Refactor or reformat? Measures.c is one of the more involved pieces in postgis, reworking it right before release is ... not something I'd necessarily do?

Komzpa commented:
From IRC discussion:
```
[02:28] <PaulRamsey[m]> just would like to confirm: did you actually refactor measures, or add in your triangle stuff and incidentally do a bunch of formatting?
[02:28] <PaulRamsey[m]> ie, is the logic largely unchanged
[02:31] <Komzpa> logic mostly unchanged
[02:34] <Komzpa> changed loops to have C99-style variable init in for, dropped if (dl->mode == DIST_MIN) after the shortcut if (dl->mode == DIST_MAX) in the top of the function, replaced homebrew atan2 with library atan2
[02:37] <Komzpa> triangles are zombie polygon-lines in our memory layout so I was building triangle functions out of polygon and line ones so had to read through them all
[02:37] <Komzpa> also fixed some of the comments
```
